src/top-bar/selection/selection-provider.js

```diff
@@ -0,0 +1,135 @@
+import { PropTypes } from '@dhis2/prop-types'
+import React, { useEffect, useReducer } from 'react'
+import { useQueryParams, StringParam } from 'use-query-params'
+import { useCurrentUser } from '../../current-user/index.js'
+import { SelectionContext } from './selection-context.js'
+
+// TODO: should take a periodCode and return a period object
+const parsePeriodCode = code => ({
+    code: code,
+    displayName: code,
+})
+
+const initialWorkflowValue = (workflows, id) => {
+    if (id) {
+        /*
+         * Auto select workflow with query param id
+         * default to empty object if `find` returns undefined in case the
+         * workflow with the id from the url is not available to the user
+         */
+        return workflows.find(workflow => workflow.id === id) || {}
+    }
+    if (workflows.length === 1) {
+        // auto-select if user only has one workflow
+        return workflows[0]
+    }
+    return {}
+}
+
+const ACTIONS = {
+    SET_OPENED_SELECT: 'SET_OPENED_SELECT',
+    CLEAR_ALL: 'CLEAR_ALL',
+    SELECT_WORKFLOW: 'SELECT_WORKFLOW',
+    SELECT_PERIOD: 'SELECT_PERIOD',
+    SELECT_ORG_UNIT: 'SELECT_ORG_UNIT',
+}
+
+const reducer = (state, { type, payload }) => {
+    switch (type) {
+        case ACTIONS.SET_OPENED_SELECT:
+            return {
+                ...state,
+                openedSelect: payload,
```

Comment:
Opinion: I think it makes more sense to make payload an object wrapper for payload values (i. e. instead of `openedSelect: payload` => `openedSelect: payload.openedSelect`. That way the payload is extensible and when reading the reducer code it'd a bit easier to know what the payload's value actually contains (i. e. `openedSelect: payload.selectName` or `openedSelect: payload.selectId`)